src/services/stats/stats.class.js

```diff
@@ -205,5 +221,105 @@ exports.Stats = class Stats extends Service {
+        const adminUser = await db.collection('users').findOne({ _id: new ObjectID(userId) });
+        if (!adminUser?.roles.includes('admin_coop')) {
+          res.status(403).send(new Forbidden('User not authorized', {
+            userId: adminUser
+          }).toJSON());
+          return;
+        }
+
+        const schema = Joi.object({
+          page: Joi.number().required().error(new Error('Le numéro de page est invalide')),
+          territoire: Joi.string().required().error(new Error('Le type de territoire est invalide')),
+          dateDebut: Joi.date().required().error(new Error('La date de début est invalide')),
+          dateFin: Joi.date().required().error(new Error('La date de fin est invalide')),
+          nomOrdre: Joi.string().error(new Error('Le nom de l\'ordre est invalide')),
+          ordre: Joi.number().error(new Error('L\'ordre est invalide')),
+        }).validate(req.query);
+
+        if (schema.error) {
+          res.status(400).send(new BadRequest('Erreur : ' + schema.error).toJSON());
+          return;
+        }
+        const page = req.query.page;
+        const territoire = req.query.territoire;
+        const dateFin = dayjs(new Date(req.query.dateFin)).format('DD/MM/YYYY');
+        const dateDebutQuery = new Date(req.query.dateDebut);
+        const dateFinQuery = new Date(req.query.dateFin);
+        const nomOrdre = req.query.nomOrdre;
+        const ordre = req.query.ordre;
+
+        //Construction des statistiques
+        let items = {};
+
+        if (territoire === 'departement') {
+          let promises = [];
+          let statsTerritoires = [];
+
+          if (nomOrdre) {
+            const ordreColonne = JSON.parse('{"' + nomOrdre + '":' + ordre + '}');
+            statsTerritoires = await db.collection('stats_Territoires').find({ 'date': dateFin })
+            .sort(ordreColonne)
+            .skip(page > 0 ? ((page - 1) * 10) : 0)
+            .limit(10).toArray();
+          } else {
+            statsTerritoires = await db.collection('stats_Territoires').find({ 'date': dateFin })
+            .skip(page > 0 ? ((page - 1) * 10) : 0)
+            .limit(10).toArray();
+          }
+
+          statsTerritoires.forEach(ligneStats => {
+
+            ligneStats.personnesAccompagnees = 0;
+            let cumulTerritoire = 0;
+
+            if (ligneStats.conseillerIds.length > 0) {
+              ligneStats.conseillerIds.forEach(conseillerId => {
+                let query = {
+                  'conseiller.$id': new ObjectID(conseillerId),
+                  'createdAt': {
+                    $gte: dateDebutQuery,
+                    $lt: dateFinQuery,
+                  }
+                };
+
+                promises.push(new Promise(async resolve => {
+                  let statsAccompagnements = await statsCras.getStatsAccompagnements(db, query);
+                  if (statsAccompagnements.length > 0) {
+                    // eslint-disable-next-line
```

Comment:
@MorganP-projects eslint-disable-next-line =>  eslint pas content sur le += ?

Reply:
C'est pour un : "Too many nested callbacks (5). Maximum allowed is 4"

Reply:
ah ok ! on pourrais préciser le max-nested-callbacks